Client/core/CDiscordRichPresence.h

```diff
@@ -69,6 +69,8 @@ class CDiscordRichPresence : public CDiscordInterface
 
     std::optional<std::tuple<std::pair<std::string, std::string>, std::pair<std::string, std::string>>> m_aButtons;
 
+    std::mutex m_threadSafetyMutex;
```

Comment:
I doesn't seem like this mutex is really needed. Is this a memory region that can be accessed from two threads at the same time?

Reply:
By adding a mutex, I wanted to further secure the initialization and updates of Rich Presence. This way, it becomes possible to effectively control access to Discord-RPC, preventing simultaneous modifications by different threads.